unstructured/metrics/evaluate.py

```diff
@@ -327,32 +360,79 @@ def measure_table_structure_accuracy(
                 ["element_col_level_content_acc", None, None, None, 0],
                 ["element_row_level_content_acc", None, None, None, 0],
             ]
-        ).reset_index()
+        ).transpose()
     else:
-        element_metrics_results = {}
-        for metric in [
-            "total_tables",
-            "table_level_acc",
-            "element_col_level_index_acc",
-            "element_row_level_index_acc",
-            "element_col_level_content_acc",
-            "element_row_level_content_acc",
-        ]:
-            metric_df = has_tables_df[has_tables_df[metric].notnull()]
-            agg_metric = metric_df[metric].agg([_mean, _stdev, _pstdev, _count]).transpose()
-            if agg_metric.empty:
-                element_metrics_results[metric] = pd.Series(
-                    data=[None, None, None, 0], index=["_mean", "_stdev", "_pstdev", "_count"]
-                )
-            else:
-                element_metrics_results[metric] = agg_metric
-        agg_df = pd.DataFrame(element_metrics_results).transpose().reset_index()
+        # filter out documents with no tables
+        having_table_df = df[df["total_tables"] > 0]
+        # compute aggregated metrics for tables
+        agg_df = having_table_df.agg(
+            {
+                "total_tables": [_mean, _stdev, _pstdev, "count"],
+                "table_level_acc": [_mean, _stdev, _pstdev, "count"],
+                "element_col_level_index_acc": [_mean, _stdev, _pstdev, "count"],
+                "element_row_level_index_acc": [_mean, _stdev, _pstdev, "count"],
+                "element_col_level_content_acc": [_mean, _stdev, _pstdev, "count"],
+                "element_row_level_content_acc": [_mean, _stdev, _pstdev, "count"],
+            }
+        ).transpose()
+        agg_df = agg_df.reset_index()
+        agg_df.columns = agg_headers
 
-    agg_df.columns = agg_headers
-    _write_to_file(
-        export_dir, "all-docs-table-structure-accuracy.tsv", _rename_aggregated_columns(df)
-    )
-    _write_to_file(
-        export_dir, "aggregate-table-structure-accuracy.tsv", _rename_aggregated_columns(agg_df)
-    )
+    _write_to_file(export_dir, "all-docs-table-structure-accuracy.tsv", df)
+    _write_to_file(export_dir, "aggregate-table-structure-accuracy.tsv", agg_df)
     _display(agg_df)
+
+
+def filter_metrics(
+    data_input: Union[str, pd.DataFrame],
+    filter_list: Union[str, List[str]],
+    filter_by: str = "filename",
+    export_filename: Optional[str] = None,
+    export_dir: str = "metrics",
+    return_type: str = "file",
+) -> Optional[pd.DataFrame]:
```

Comment:
I think we need a nice docstring here describing what this function is exactly doing and what each of the parameters means as well as the returned value.